src/mpyl/steps/deploy/dagster.py

```diff
@@ -0,0 +1,104 @@
+"""
+Step to deploy a dagster user code repository to k8s
+"""
+from logging import Logger
+from typing import List
+
+
+from .k8s import helm, get_key_of_config_map, rollout_restart_deployment, cluster_config
+from ..models import RunProperties
+from .. import Step, Meta, ArtifactType, Input, Output
+from ...project import Stage, Project, Target, get_env_variables
+from ...stages.discovery import find_deploy_set
+from ...utilities.repo import RepoConfig
+
+
+class DeployDagster(Step):
+
+    def __init__(self, logger: Logger) -> None:
+        super().__init__(logger, Meta(
+            name='Dagster Deploy',
+            description='Deploy a dagster user code repository to k8s',
+            version='0.0.1',
+            stage=Stage.DEPLOY
+        ), produced_artifact=ArtifactType.NONE, required_artifact=ArtifactType.DOCKER_IMAGE)
+
+    def __to_user_code_deployment(self, project: Project, run_properties: RunProperties):
+        env_variables = get_env_variables(project, run_properties.target)
+        name_prefix = f'pr-{run_properties.versioning.pr_number}-' if run_properties.target == Target.PULL_REQUEST else ''
+        return {'deployments': [{
+            'dagsterApiGrpcArgs': [
+                "--python-file",
+                project.dagster.repo
+            ],
+            'env': env_variables,
+            'envSecrets': [],
+            'image': {
+                'pullPolicy': 'Always',
+                'imagePullSecrets': [
+                    {
+                        'name': 'bigdataregistry'
+                    }
+                ],
+                'tag': run_properties.versioning.identifier,
+                'repository': f'bigdataregistry.azurecr.io/{project.name}'
+            },
+            'includeConfigInLaunchedRuns': {
+                'enabled': True
+            },
+            'name': f'{name_prefix}{project.name}',
+            'port': 3030
+        }]}
```

Comment:
export it into something like `to_spark_body()`

Reply:
hard coded names of the registry -> use config

Reply:
didnt find any field for `imagePullSecrets` tho. ok to keep it hard coded?

Reply:
I meant like use the registry name from the mpyl_config.yaml

Reply:
oh yes, i also found that and applied it, but then i noticed that the `imagePullSecret` will probably stay hardcoded